src/coreclr/dlls/mscordac/CMakeLists.txt

```diff
@@ -31,9 +31,9 @@ else(CLR_CMAKE_HOST_WIN32)
     # Add dependency on export file
     add_custom_target(mscordaccore_exports DEPENDS ${EXPORTS_FILE})
 
-    if(CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS OR CLR_CMAKE_HOST_HAIKU)
+    if(CLR_CMAKE_HOST_APPLE OR CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS OR CLR_CMAKE_HOST_HAIKU)
         generate_exports_file(${DEF_SOURCES} ${EXPORTS_FILE})
-    endif(CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS OR CLR_CMAKE_HOST_HAIKU)
+    endif(CLR_CMAKE_HOST_APPLE OR CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS OR CLR_CMAKE_HOST_HAIKU)
 
     if(CORECLR_SET_RPATH AND CLR_CMAKE_HOST_OSX AND CLR_CMAKE_HOST_ARCH_ARM64)
```

Comment:
Has CLR_CMAKE_HOST_OSX been completed replaced by CLR_CMAKE_HOST_APPLE? If so, then the RPATH in the DAC isn't be added.


@hoyosjs

Reply:
Never mind.  CLR_CMAKE_HOST_OSX is still defined if not maccatalyst.

Reply:
The idea is to use same parameters for all Apple platforms (macOS, iOS, tvOS) whenever possible. RPATH may still need some tweaks for each platform due to different bundle structure on macOS/MacCatalyst and iOS/tvOS.

Reply:
I'll need to do another pass over the `CLR_CMAKE_HOST_OSX` usages. The RPath ones are likely wrong for some platforms. We don't produce packages for those platforms yet, so the code has never been tested. As for other occurrences - `createdump` is macOS only (but maybe MacCatalyst too?), guards in test code are likely unnecessary because `_DARWIN_C_SOURCE` is now set globally (to be verified).